server/app/controllers/applicant/ApplicantProgramReviewController.java

```diff
@@ -165,6 +178,20 @@ private CompletionStage<Result> submitInternal(
                       routes.ApplicantProgramReviewController.review(applicantId, programId);
                   return redirect(reviewPage).flashing("error", errorMsg);
                 }
+                if (cause instanceof ApplicationNotEligibleException) {
```

Comment:
thought: how did the official java way to do async programming not get feature parity with `catch` for handling typed exceptions? smh

Reply:
Clean use of Futures in java is def a weakness for me, not sure if there's a better way to manage this.  We also have to make them Runtime exceptions even if they maybe shouldn't it seems.

Reply:
I don't know of a better way to handle this, I think it's an oversight in the language design